### PR TITLE
Working and refactored

### PR DESCRIPTION
**Pull request by hookom**
App.js should be reviewed in its entirety in this PR. Copy in master was not working and was not reviewed, just uploaded in a non-working state when we created the repo.

All other files are auto-generated by rally-app-builder.

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -10,132 +10,165 @@ Ext.define('CustomApp', {
       oldTestId = this.add({
         xtype: 'textfield',
         name: 'oldTest',
+        emptyText: '1234',
```
Comment (mjvcallibrity) [inline App.js]: Perhaps change this `ex. 1234`?

File: App.js [hunk 2]
```diff
@@ -10,132 +10,165 @@ Ext.define('CustomApp', {
...
+        submitEmptyText: false,
+        maskRe: /\d/,
         fieldLabel: 'OLD Test Case ID',
-        allowBlank: false,
       });
 
       newUserStoryId = this.add({
         xtype: 'textfield',
         name: 'newStory',
+        emptyText: '5678',
```
Comment (mjvcallibrity) [inline App.js]: Also change this to `ex. 5678`?

File: App.js [hunk 3]
```diff
@@ -10,132 +10,165 @@ Ext.define('CustomApp', {
...
+        submitEmptyText: false,
+        maskRe: /\d/,
         fieldLabel: 'NEW User Story ID',
-        allowBlank: false,
       });
 
       submitButton = this.add({
         xtype: 'rallybutton',
         text: 'Next',
```
Comment (mjvcallibrity) [inline App.js]: Change this text to something more descriptive. Right now it implies you are going to navigate to a different view , but instead the two fields reset to their placeholder text.

File: App.js [hunk 4]
```diff
@@ -10,132 +10,165 @@ Ext.define('CustomApp', {
...
-                  value: 'TC'+tcId
-              }
-          ]
-        });
+        if(!tcFormatId || !usFormatId) {
+            alert("Please provide both values.");
+        }
+        else {
+
+            var tcId_p = retrieveObjectId('testcase', tcFormatId);
```
Comment (mjvcallibrity) [inline App.js]: What is the `_p` mean? Could these variables be renamed to something more descriptive?

**Comment by richardbarker** [comment]
I like the changes you made since I last looked at it with you.  Well done.

File: App.js [hunk 5]
```diff
@@ -10,132 +10,165 @@ Ext.define('CustomApp', {
...
+                // Once we have the objectID of the new testcase, load its record.
+                .then(function(objId) {
+                    return loadTestCaseRecord(objId);
+                })
+                // Once we have the new test record, update and save it.
+                .then(function(newTestCase) {
+                    newTestCase.set('Name', newTestCase.data.Name.substring(9));
+
+                    var newTextToPrepend = "(Copied from " + oldTestCase.data.FormattedID + ")\n";
+                    var descrip = newTestCase.data.Description;
```
Comment (mjvcallibrity) [inline App.js]: Does much get saved my abbreviating description to descrip?

File: App.js (hunk 4 above)
Comment (hookom) [inline App.js]: I used _p for promise-holding variables. I didn't know the convention (or if there was one) :)

File: App.js (hunk 5 above)
Comment (hookom) [inline App.js]: nope

**Pull request merged**